Implement review comments for #16

JanKanis opened the pull request:
- Use logging
- pyflake8'ify
- better messages when lowering rendering thread priority fails

Pylint also wants me to give every class/method a docstring, which I didn't do. 

I'll (try to) have a look at your preferences screen tomorrow.

bittner commented:
Oh, I'm sorry you also started working on the same thing (logging) and I noticed it only now!  :worried: 

You may have seen that I have refactored the code quite a bit. At the end, I also added the logging in a way very similar as you attempted it.

Maybe you want to rebase your branch, or start the Flake8 and Pylint fixes from scratch. I know this feels like a shame, I'm sorry! :pensive:

bittner commented:
There is a single issue open for the preference dialog:

How to move the active plugin rendering pane from the bottom panel to the side panel, and vice-versa, while gedit is running. I have the [code that is not working](https://github.com/bittner/gedit-reST-plugin/blob/master/reST/config.py#L90-L93) commented out. The reference to the parent window (gedit's main window) is `None` for some reason, and only set correctly when gedit is exiting.

We need to find out why, and how to fix that. Maybe you have an idea. I had planned to post a question later today on the [GNOME Discourse](https://discourse.gnome.org/) (discussion forum) about that if I make no progress on the issue myself.

JanKanis commented:
@bittner I updated the PR to be on top of your current master. It's mostly the same changes as the previous commit, except you already changed to using logger.

A few notes: 
- calling `logging.basicConfig` configures logging application-wide, so calling that from a plugin is not a good idea. Also setting a format that does not include the module name makes it very hard to find out where a message comes from if other plugins/modules also emit logging messages. So I removed the call, and set the log level on the logger object.
- You shouldn't format strings passed to logging methods, that negates the benefit of not having the overhead of formatting when a message is not printed.
